File: purchase/src/main/java/com/purchase/purchase/model/Product.java
package com.purchase.purchase.model;

import java.time.LocalDate;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Version;
import lombok.EqualsAndHashCode;
import lombok.experimental.FieldNameConstants;
import org.hibernate.annotations.Check;
import org.springframework.scheduling.annotation.Async;

@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Entity

public class Product {
    @Id
    @GeneratedValue
    private Integer id;
    private String productName;
    @Check(constraints = "qty >= 0")
    private Integer qty;
    private Integer price;  
    private LocalDate startDate;
    private LocalDate endDate;
    private LocalDate todayDate;
    private Integer card_number;  
    private Integer cvv;
    
}

File: angular-17-jwt-auth-master/src/app/model/Product.ts
export interface Product {
        id: number,
        filteredData: string,
        productName: string,
        qty: any,
        price: number,
        [key: string]: any,
        addControl: any,
        displayedColumns: any,
        startDate: Date,
        endDate: Date,
        page:any,
        size:any;
        card_number:any;
        cvv:any;
}